src/core_plugins/kibana/public/management/sections/indices/add_data_steps/parse_csv_step/parse_csv_step.js

```diff
@@ -67,27 +68,22 @@ modules.get('apps/management')
                   return;
                 }
                 if (row === 1) {
-                  // Collect general information on the first pass
-                  if (results.meta.fields.length > _.uniq(results.meta.fields).length) {
-                    this.formattedErrors.push('Column names must be unique');
-                  }
-
-                  let hasEmptyHeader = false;
-                  _.forEach(results.meta.fields, (field) => {
-                    if (_.isEmpty(field)) {
-                      hasEmptyHeader = true;
+                  // Check for header errors on the first row
+                  let errors = validateHeaders(results.meta.fields);
+                  _.forEach(errors, (error) => {
+                    if (error.type === 'duplicate') {
+                      this.formattedErrors.push(`Columns at positions [${error.positions}] have duplicate name "${error.fieldName}"`);
+                    } else if (error.type === 'blank') {
+                      this.formattedErrors.push(`Columns at positions [${error.positions}] must not be blank`);
```

Comment:
Yeah I was debating with myself too. I decided to keep the error messages in `parse_csv_step` because to keep the error messages and warning messages under the same file makes more sense and readable. Plus, it makes it easier to unit test the `validateHeaders` lib.